addons/sosivio/enable

```diff
@@ -2,58 +2,192 @@
 CURRENT_DIR=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)
 
+KUBECTL="$SNAP/microk8s-kubectl.wrapper"
+HELM="$SNAP/microk8s-helm3.wrapper"
+
+
+
+# script vars
 BLUE='\033[0;34m'
 NC='\033[0m'
 BOLD='\033[1m'
-echo ""
-echo -e "${BLUE}Enabling Sosivio${NC}"
-echo ""
-
-"$SNAP/microk8s-enable.wrapper" dns
-"$SNAP/microk8s-enable.wrapper" helm3
 
-
-KUBECTL="$SNAP/microk8s-kubectl.wrapper"
-HELM="$SNAP/microk8s-helm3.wrapper"
 NAMESPACE_SOSIVIO="sosivio"
 DASHBOARD_SERVICE="dashboard-lb"
 
+HELM_REPO_URL="https://helm.sosiv.io"
+HELM_REPO_NAME="sosivio"
+HELM_VERSION=""
+WAIT_SECONDS=300
+DEBUG=0
+
+usage () {
+  echo '
+Sosivio Plugin for Microk8s
+
+For Additional infromation about Sosivio please infer our website:
+https:/sosiv.io
+Or The official Documentation:
+https://docs.sosiv.io
+
+Usage:
+  microk8s enable sosivio
+
+Flags:
+    -d         enable debug prints
+    -v string  install a specific helm chart version (default to latest)
+    -u string  the URL of the helm repo to fetch the charts from (default "$HELM_REPO_URL")
+    -n string  the name of the helm repo name to add (default "$HELM_REPO_NAME") 
+    -w int     amount of time in seconds to wait for sosivio to become active (default "$WAIT_SECONDS")       
+'
+}
+
+# opt_error will print $1 input, Help and will exit with code 1
+opt_error() {
+  echo $1
+  usage
+  exit 1
+}
+
+
+print_debug() {
+  if [ $DEBUG -eq 1 ]; then
+    echo "DEBUG: $1"
+  fi
+}
 
+# install microk8s addons that are a prerequisite to Sosivio addon
+install_addons_prereq () {
+  addons=("dns" "helm3")
+  print_debug "enabling prerequisites addons..."
+  for addon in ${addons[@]}; do
+    print_debug "  enabling $addon"
+    "$SNAP/microk8s-enable.wrapper" $addon &> /dev/null
+  done
+  print_debug "all prerequisites addons installed successfuly"
+  # "$SNAP/microk8s-enable.wrapper" dns
+  # "$SNAP/microk8s-enable.wrapper" helm3
+}
 
-$HELM repo add sosivio https://helm.sosiv.io > /dev/null 2>&1
-$HELM repo update > /dev/null 2>&1
-$KUBECTL create ns $NAMESPACE_SOSIVIO > /dev/null 2>&1 || true
-$HELM upgrade -n $NAMESPACE_SOSIVIO --install \
-  sosivio sosivio/sosivio --version 1.5.3 \
-  --set expose=LoadBalancer > /dev/null 2>&1 || true
+add_helm_repo () {
+  print_debug "adding repo $HELM_REPO_NAME $HELM_REPO_URL"
+  $HELM repo add $HELM_REPO_NAME $HELM_REPO_URL > /dev/null 2>&1
+  print_debug "updating repo..."
+  $HELM repo update $HELM_REPO_NAME > /dev/null 2>&1
+  
+}
 
-echo ""
-echo "Waiting for Sosivio to become active..."
-echo "It might take a while if your network connection is slow."
+install_soivio () {
 
-WATCH="$KUBECTL rollout status --watch"
-until $WATCH deploy -n ${NAMESPACE_SOSIVIO} sosivio-dashboard --watch > /dev/null 2>&1
-do
-  sleep 1
+  echo ""
+  echo -e "${BLUE}Enabling Sosivio${NC}"
+  echo ""
+
+  print_debug "  creating $NAMESPACE_SOSIVIO namespace"
+  $KUBECTL create ns $NAMESPACE_SOSIVIO > /dev/null 2>&1 || true
+
+  HELM_CMD="$HELM upgrade --install -n $NAMESPACE_SOSIVIO \
```

Comment:
Could we please make sure we install a specific version of sosivio so we do not have this problem in the future? This version should also be set in the top level addons.yaml file describing the addon.

Reply:
Any update on this @DanArlowski ?